YoCode/ProcessRunner.cs

```diff
@@ -133,13 +136,36 @@ private void ErrorDataReceived(object sender, DataReceivedEventArgs e)
             errorOutput.Add(e.Data);
         }
 
-        public void KillLiveProcess()
+        private static void KillLiveProcess(Process p)
         {
             if (!p.HasExited)
             {
                 p.Kill();
                 p.Dispose();
             }
         }
+
+        private static void FindAndKillChildProcesses(int pid)
+        {
+            using (var mos = new ManagementObjectSearcher(string.Format(CultureInfo.InvariantCulture,
+                  "Select * From Win32_Process Where ParentProcessID={0}", pid)))
+            {
+                foreach (var mo in mos.Get())
+                {
+                    try
+                    {
+                        var processById = Process.GetProcessById(Convert.ToInt32(mo["ProcessID"]));
```

Comment:
When I say "make `FindAndKillChildProcesses` recursive" I mean that it could call itself here, passing `processByID.Id`.